### DISCUSSION TITLE
Connecting via VNC?

### DISCUSSION THREAD

**Discussion by vadi2** [community/community#25453] [Codespaces]
<p>My application runs via VNC, how can I connect to it?</p>
<p>Trying to connect to the &lt;uuid&gt;-&lt;port&gt;.apps.codespaces.githubusercontent.com does establish a successful VNC connection, but the handshake does not work.</p>

**Comment by jkeech** [comment, marked as answer]
<p><a class="mention" href="/vadi2">@vadi2</a>, if you connect from VS Code on the desktop and run the app in the Codesapce, it will show up on localhost:port and then your VNC connection should work. The web-based port forwarding in the browser does not handle all protocols today and it mainly focused on HTTP(S), but when you run from the desktop on localhost, it forwards all traffic over without the additional proxy.</p>

**Comment by Chuxel** [comment]
<p><a class="mention" href="/vadi2">@vadi2</a> In addition to what <a class="mention" href="/jkeech">@jkeech</a> mentions, noVNC is an option to provide a web based client installed in your container.  We’ve got a desktop script here that uses it: <a href="https://github.com/microsoft/vscode-dev-containers/blob/2fbbbc61b3ae414219a51b64399375933a5d9d36/script-library/docs/desktop-lite.md" rel="noopener nofollow ugc">https://github.com/microsoft/vscode-dev-containers/blob/2fbbbc61b3ae414219a51b64399375933a5d9d36/script-library/docs/desktop-lite.md</a></p>
<p>Or you can just follow the directions at <a href="https://novnc.com/info.html" rel="noopener nofollow ugc">https://novnc.com/info.html</a>  It works surprisingly well.</p>

**Comment by vadi2** [comment]
<p>Unless I set it up wrong, it did not seem to work:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8e8a8f975909ee5c5358a64c5c8a9750ad71a245.png" data-download-href="/uploads/short-url/kkYFiKRSmlNTaF68UadVC9umdcp.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/110988/181091184-5256c54d-04b4-41ed-9d30-1b851678c1ff.png" alt="image" data-base62-sha1="kkYFiKRSmlNTaF68UadVC9umdcp" width="690" height="390" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8e8a8f975909ee5c5358a64c5c8a9750ad71a245_2_690x390.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/8e8a8f975909ee5c5358a64c5c8a9750ad71a245_2_1035x585.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/8e8a8f975909ee5c5358a64c5c8a9750ad71a245.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8e8a8f975909ee5c5358a64c5c8a9750ad71a245_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1315×744 124 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jkeech** [comment]
<p><a class="mention" href="/vadi2">@vadi2</a>, are you running the VS Code desktop app when connected to the Codespace, or via the browser? Are you forwarding the port 5900 locally? If you look in the Remote Explorer view let you can see which ports are forwarded.</p>

**Comment by vadi2** [comment]
<p>VS Code is running locally on my Ubuntu, as I understood in your post.</p>
<blockquote>
<p>Are you forwarding the port 5900 locally?</p>
</blockquote>
<p>I was not… but now that I did, it works!</p>
<p><a class="mention" href="/chuxel">@Chuxel</a> clued onto what I was hoping to do though, now I am trying the browser-based version. Would be great if people can code on our <a href="https://github.com/Mudlet/Mudlet" rel="noopener nofollow ugc">MUD client</a> with Codespaces :slight_smile:</p>

**Comment by jkeech** [comment]
<p>Glad to hear you got it working on the desktop! We’ve heard that some folks need to use other desktop apps (which don’t always have browser-based equivalents), so the desktop app port forwarding is the way to go in those cases. But if you have a way to keep everything in the browser, that is often more convenient/preferred.</p>

**Comment by vadi2** [comment]
<p>Just an FYI, I’ve put this together into an article: <a href="https://www.mudlet.org/2020/10/develop-run-desktop-apps-in-the-cloud-github-codespaces" rel="noopener nofollow ugc">https://www.mudlet.org/2020/10/develop-run-desktop-apps-in-the-cloud-github-codespaces</a></p>

**Comment by Chuxel** [comment]
<p>Yeah as a tip, the default for the desktop script uses port 5901 instead of 5900 since 5900 can already be in use on your local machine (particularly on Linux).  The SSH script uses 2222 for the same reason.</p>

**Comment by samdenty** [comment]
It would be really cool if codespaces port forwarding could support tcp communication so i can connect to the vnc server that’s built into codespaces, on ipados using realvnc viewer app. The no-vnc app doesn’t have proper integration with clipboard, keyboard shortcuts etc. on ios. Also it sometimes gets killed to free up memory so i always have to reconnect and enter vscode as password which is annoying
